### PR TITLE
Renew git autoupdate config in autosize.js, close #5067

### LINKED ISSUE #5067: [Request] Library Autosize v3.0.8
From: https://github.com/jackmoore/autosize

### PR DESCRIPTION
**Pull request by Piicksarn**

### CODE REVIEW

**Comment by Piicksarn** [comment]
@Amomo  would you please help me checking the commits?
These commits renew the information in package.json of autosize lib,and renew the git autoupdate config in it.
Thank you！

**Comment by Amomo** [comment]
I think that the first commit message 0d7de45 is not very clear.
You may add the library's name and modified file's name (if you just modify one file, like package.json) into it, this action will make your commit more readable to entire repository.
And the second commit 9640cb4, I believe that separating it into two commits will be better.

**Comment by Piicksarn** [comment]
@Amomo thank you!
the commits become much more clear now.
i think it is ok now~how do you think? :blush:

**Comment by Amomo** [comment]
Great! It looks much better :D

**Comment by Piicksarn** [comment]
@Peter

File: ajax/libs/autosize.js/package.json [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+  },
+  "description": "Autosize is a small, stand-alone script to automatically adjust textarea height to fit text.",
   "homepage": "http://www.jacklmoore.com/autosize",
   "keywords": [
-    "autosize",
     "form",
     "textarea",
-    "ui",
-    "jQuery"
+    "ui"
```
Comment (PeterDaveHello) [inline ajax/libs/autosize.js/package.json]: What's the reason why to remove these keywords?

File: ajax/libs/autosize.js/package.json [hunk 2]
```diff
@@ -24,19 +26,13 @@
...
+  "licenses": "MIT",
   "autoupdate": {
     "source": "git",
     "target": "git://github.com/jackmoore/autosize.git",
-    "basePath": "",
+    "basePath": "dist",
     "files": [
-      "jquery.autosize.js",
-      "jquery.autosize.min.js"
+      "autosize*.js"
```
Comment (PeterDaveHello) [inline ajax/libs/autosize.js/package.json]: Use **/* in dist.

File: ajax/libs/autosize.js/package.json (hunk 1 above)
Comment (Piicksarn) [inline ajax/libs/autosize.js/package.json]: this correction is referenced from the latest package.json of autosize lib

**Comment by Piicksarn** [comment]
have it corrected~

File: ajax/libs/autosize.js/package.json (hunk 1 above)
Comment (PeterDaveHello) [inline ajax/libs/autosize.js/package.json]: I don't think this is a correction.

Comment (Piicksarn) [inline ajax/libs/autosize.js/package.json]: So..it's needless to make these changes in keywords?

Comment (PeterDaveHello) [inline ajax/libs/autosize.js/package.json]: Of course.

**Comment by Piicksarn** [comment]
* have meaningful information corrected.
* have files in autoupdate corrected.

**Comment by Piicksarn** [comment]
@Amomo how do you think about these correction :^))?~

**Comment by Amomo** [comment]
I think you have made it. The corrections are ok.

File: ajax/libs/autosize.js/package.json [hunk 3]
```diff
@@ -1,9 +1,13 @@
 {
   "name": "autosize.js",
-  "filename": "jquery.autosize.min.js",
-  "version": "1.18.18",
-  "author": "Jack Moore",
-  "description": "Automatically adjust textarea height based on user input.",
+  "filename": "autosize.min.js",
```
Comment (PeterDaveHello) [inline ajax/libs/autosize.js/package.json]: Should talk about this change in commit message ...

**Comment by PeterDaveHello** [comment]
Commit's parent was too old.

**Comment by Piicksarn** [comment]
@Amomo i make the detail in the commits, and i need your help for checking again...
Thank you!!

**Comment by Amomo** [comment]
Sorry for my delay! I think it's ok~ 
@PeterDaveHello , Does this branch look too old?

**Comment by Piicksarn** [comment]
@Amomo, It's nothing:))~I think it's necessary to rebase to the master, commit's parent is too old!

**Comment by PeterDaveHello** [comment]
So why don't you just do it?

**Comment by Piicksarn** [comment]
@Amomo ~I have rebased commits to the master, and need your help for checking:^D

**Comment by Amomo** [comment]
ok! @PeterDaveHello I think it matches the latest branch.

**Comment by PeterDaveHello** [comment]
Commit message of ddf6397 is too long.

**Comment by PeterDaveHello** [comment]
22a0c05 in unnecessary.

**Comment by Piicksarn** [comment]
@PeterDaveHello ~I don't understand about why 22a0c05 in unnecessary...
the library had became using autosize.min.js to replace jquery.autosize.min.js, and i updated the lib to the latest version with autosize.min.js, so that the version released in the future can update with autosize.min.js.

Is there other way to make autosize.min.js to replace jquery.autosize.min.js?~

**Comment by PeterDaveHello** [comment]
If a lib can use auto-update, you should leave it alone with real files, maintainers will handle it.

**Comment by Piicksarn** [comment]
@PeterDaveHello i got it! Thank you! I will do in right way next time when it comes the same situation.
@Amomo thanks a lot for this pr and i have them corrected.

**Pull request merged**